Source/OCMock/OCMockObject.m

```diff
@@ -261,50 +284,75 @@ - (void)forwardInvocation:(NSInvocation *)anInvocation
     }
     @catch(NSException *e)
     {
-        [exceptions addObject:e];
+        @synchronized(exceptions)
+        {
+            [exceptions addObject:e];
+        }
         [e raise];
     }
 }
 
 - (BOOL)handleInvocation:(NSInvocation *)anInvocation
 {
-    [invocations addObject:anInvocation];
-
+    @synchronized(invocations)
+    {
+        // When the method has a char* argument we do not retain the arguments. This makes it possible
+        // to match char* args literally and with anyPointer. Not retaining the argument means that
+        // in these cases tests that use their own autorelease pools may fail unexpectedly.
+        if(![anInvocation hasCharPointerArgument])
+            [anInvocation retainArguments];
+        [invocations addObject:anInvocation];
+    }
+    
```

Comment:
Keep in mind that -retainArguments will also retain the target, plus any future target (if this invocation is used with -invoke or -invokeWithTarget:).  So, if the target is self (unsure if that is possible in this context) then this is creating a retain loop and the mock won't ever dealloc.

If the target is never used, it might be worthwhile setting the target to nil before calling -retainArguments on any stored NSInvocation.